mypy/typeanal.py

```diff
@@ -413,7 +413,8 @@ def visit_callable_type(self, t: CallableType, nested: bool = True) -> Type:
                 variables = self.bind_function_type_variables(t, t)
             ret = t.copy_modified(arg_types=self.anal_array(t.arg_types, nested=nested),
                                   ret_type=self.anal_type(t.ret_type, nested=nested),
-                                  fallback=t.fallback or self.named_type('builtins.function'),
+                                  fallback=(t.fallback if t.fallback.type
```

Comment:
This pattern is a bit confusing, I would add a little remainder here that `FakeInfo` is `False`y. (maybe also below?)